pkg/metrics/statsd/cosmosdb/metrics.go

```diff
@@ -62,6 +62,12 @@ func (t *tracerRoundTripper) RoundTrip(req *http.Request) (resp *http.Response,
 				"verb": req.Method,
 				"path": path,
 			})
+		} else {
```

Comment:
@mjudeikis convinced me that this function was OK back in the day but I am still doubtful about it.

If `t.tr.RoundTrip(req)` returns an error, then the `defer()` runs, then `ru, err = strconv.ParseFloat(requestCharge, 64)` runs successfully, does that mean that the original error gets overwritten and the end user never sees it?

I think we might need to use a different variable name from `err` throughout the `defer()` - and here we need to test the *original* error to decide whether to increment client.cosmosdb.errors or not.

@kwoodson please can you investigate / fix as part of this PR?

Reply:
@jim-minter @mjudeikis I have renamed the `err` variable inside of the `defer` and then I have setup the tests to verify the emits are being called.

Is this more in line with what you wanted?

Reply:
Im a little bit confused about the code bellow (might be lack of coffee).
I think readability of this function is terrible (my fault).
Maybe we should not do empty return with `return t.tr.RoundTrip(req)`

and do something like : 
```
defer(){
// RU cound parsing and checking
}
resp, err:= RoundTrip(req)
if err!=nil{
// emit CosmosDB error metric
}
return resp, err
```

As now this empty return and enourmous defer is confusing a lot!